pkgs/by-name/ta/tabby-terminal/package.nix

```
@@ -0,0 +1,72 @@
+{ stdenv, fetchurl, pkgs, lib, alsa-lib }:
+
+stdenv.mkDerivation rec {
+  name = "tabby-terminal";
+  version = "1.0.207";
+  src = fetchurl {
+    url = "https://github.com/Eugeny/tabby/releases/download/v${version}/tabby-${version}-linux-x64.deb";
+    hash = "sha256-PTTkL3+mYb7KM8fDUmgCuAF2lU88fYOstGWp/O5WZas=";
+  };
+
+  nativeBuildInputs = with pkgs; [
+    dpkg
+    makeWrapper
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    dpkg -x $src .
+
+    runHook postUnpack
+  '';
+
+  buildInputs = with pkgs; [
+    glib
+    nss
+    nspr
+    atk
+    cups
+    dbus
+    libdrm
+    gtk3
+    pango
+    cairo
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    libxkbcommon
+    mesa
+    expat
+    xorg.libxcb
+    alsa-lib
+    libGL
+    libsecret
+    musl
+  ];
+
+  postInstall = ''
+    mkdir -p $out/{opt,bin}
+    cp -r opt $out
+    wrapProgram "$out/opt/Tabby/tabby" --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}} --no-sandbox" --set LD_LIBRARY_PATH=${lib.makeLibraryPath buildInputs}
```

```suggestion
    wrapProgram "$out/opt/Tabby/tabby" --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}} --no-sandbox"
```

Don't prioritize LD_LIBRARY_PATH. Try using `autoPatchelfHook` instead. See https://nixos.org/manual/nixpkgs/unstable/#setup-hook-autopatchelfhook (You have to look carefully how `buildInputs` works in this case, and use some `runtimeDependencies` if that doesn't work)

---

I tried this:
```nix
{
  stdenv,
  fetchurl,
  pkgs,
  lib,
  alsa-lib,
}:

stdenv.mkDerivation rec {
  name = "tabby-terminal";
  version = "1.0.207";
  src = fetchurl {
    url = "https://github.com/Eugeny/tabby/releases/download/v${version}/tabby-${version}-linux-x64.deb";
    hash = "sha256-PTTkL3+mYb7KM8fDUmgCuAF2lU88fYOstGWp/O5WZas=";
  };

  nativeBuildInputs = with pkgs; [
    dpkg
    autoPatchelfHook
  ];

  buildInputs = with pkgs; [
    glib
    nss
    nspr
    atk
    cups
    dbus
    libdrm
    gtk3
    pango
    cairo
    xorg.libX11
    xorg.libXcomposite
    xorg.libXdamage
    xorg.libXext
    xorg.libXfixes
    xorg.libXrandr
    libxkbcommon
    mesa
    expat
    xorg.libxcb
    alsa-lib
    libGL
    libsecret
    musl
  ];

  runtimeDependencies = buildInputs;

  postInstall = ''
    mkdir -p $out/{opt,bin}
    cp -r opt $out
    # wrapProgram "$out/opt/Tabby/tabby" --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}} --no-sandbox"
    cp -r usr/* $out

    mv $out/opt/Tabby/tabby $out/bin/tabby

    substituteInPlace $out/share/applications/tabby.desktop \
      --replace "/opt/Tabby/tabby" tabby
  '';

  meta = with lib; {
    homepage = "https://tabby.sh";
    description = "A terminal for a more modern age";
    license = licenses.mit;
    maintainers = with maintainers; [ ChocolateLoverRaj ];
    mainProgram = "tabby";
    platforms = platforms.linux;
    downloadPage = "https://github.com/Eugeny/tabby/releases/latest";
  };
}
```
When I run it I get this error:
```bash
[rajas@nixos:~/Documents/nixpkgs]$ ./result/bin/tabby 
[0516/104608.334415:ERROR:icu_util.cc(240)] Invalid file descriptor to ICU data received.
Trace/breakpoint trap (core dumped)
```

---

Maybe ico is missing?

---

@SuperSandro2000 should I try adding [`xorg.ico`](https://search.nixos.org/packages?channel=unstable&show=xorg.ico&from=0&size=50&sort=relevance&type=packages&query=ico)?